src/Type.php

```diff
@@ -436,6 +436,78 @@ public function exists($conditions)
     }
 
     /**
+     * Persists a list of entities based on the fields that are marked as dirty and
+     * returns the same entity after a successful save or false in case
+     * of any error.
+     * Triggers the `Model.beforeSave` and `Model.afterSave` events.
+     * ## Options
+     * - `checkRules` Defaults to true. Check deletion rules before deleting the record.
+     *
+     * @param array An array of entities
+     * @param array $options An array of options to be used for the event
+     * @return bool
+     */
+    public function saveMany($entities, $options = [])
+    {
+        $options += ['checkRules' => true];
+        $options = new ArrayObject($options);
+
+        $documents = [];
+
+        foreach ($entities as $key => $entity) {
+            if (!$entity instanceof EntityInterface) {
+                throw new RuntimeException(sprintf(
+                    'Invalid items in the list. Found `%s` but expected `%s`',
+                    is_object($entity) ? get_class($entity) : gettype($entity),
+                    EntityInterface::class
+                ));
+            }
+
+            $this->dispatchEvent('Model.beforeSave', [
+                'entity' => $entity,
+                'options' => $options
+            ]);
```

Comment:
Shouldn't you check that this event wasn't stopped? The ORM will abort saves when `Model.beforeSave` is cancelled.